src/PSR7Session/Storage/FileStorage.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace PSR7Session\Storage;
+
+use PSR7Session\Id\Factory\SessionIdFactoryInterface;
+use PSR7Session\Id\SessionIdInterface;
+use PSR7Session\Session\DefaultSessionData;
+use PSR7Session\Session\SessionInterface;
+use PSR7Session\Session\StorableSessionInterface;
+
+class FileStorage implements StorageInterface
+{
+    /** @var string */
+    private $directory;
+    /** @var SessionIdFactoryInterface */
+    private $idFactory;
+
+    public function __construct(string $directory, SessionIdFactoryInterface $idFactory)
+    {
+        $this->directory = $directory;
+        $this->idFactory = $idFactory;
+    }
+
+    public function save(StorableSessionInterface $session)
+    {
+        file_put_contents($this->buildPath($session->getId()), json_encode($session));
```

Comment:
This is asking for race conditions when writing to the same file at the same time. I'd suggest using LOCK_EX as the third parameter to `file_put_contents()`